### PR TITLE
New Adapter: Gothamads

### PR DESCRIPTION
**Pull request by gotham-ads**
Hi, please take a look

Related PR in [prebid.github.io](https://github.com/prebid/prebid.github.io) - https://github.com/prebid/prebid.github.io/pull/4692

### CODE REVIEW

**Comment by gargcreation1992** [comment]
@gotham-ads PR checks are failing. Please check.

File: static/bidder-info/gothamads.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+endpoint: "http://us-e-node1.gothamads.com/?pass={{.AccountID}}"
+maintainer:
+  email: "support@gothamads.com"
```
Comment (onkarvhanumante) [inline static/bidder-info/gothamads.yaml:3]: @gotham-ads , email is being sent from prebid team to verify email address. Requesting  to respond back to complete verification.

File: static/bidder-info/gothamads.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+endpoint: "http://us-e-node1.gothamads.com/?pass={{.AccountID}}"
```
Comment (onkarvhanumante) [inline static/bidder-info/gothamads.yaml:1]: verified that endpoint is reachable 

```
~/prebid-repos/fork-repos/prebid-server (master) » curl -i http://us-e-node1.gothamads.com
HTTP/1.1 200 OK
Server: nginx
Date: Tue, 04 Jul 2023 07:58:41 GMT
Content-Type: application/json; charset=UTF-8
Content-Length: 14
Connection: keep-alive
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=63072000; includeSubdomains; preload
X-Frame-Options: DENY

no bid request
```

File: adapters/gothamads/gothamads.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	}
+
+	return headers
+}
+
+func (a *adapter) MakeRequests(openRTBRequest *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) (requestsToBidder []*adapters.RequestData, errs []error) {
+	impExt, err := getImpressionExt(&openRTBRequest.Imp[0])
+	if err != nil {
+		return nil, []error{err}
+	}
```
Comment (onkarvhanumante) [inline adapters/gothamads/gothamads.go:60]: @gotham-ads, is bid from bidder server side will be only for first imp? `imp[0]` is used to build request header. Does this implies that remaining imps won't be considered?

If yes then this behaviour should be mentioned in bidder docs

File: adapters/gothamads/gothamads.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+	openRTBRequest.Imp[0].Ext = nil
```
Comment (onkarvhanumante) [inline adapters/gothamads/gothamads.go:62]: @gotham-ads, for our understanding. Why is `imp[0].Ext` set to nil

File: adapters/gothamads/gothamads.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+				return openrtb_ext.BidTypeVideo, nil
+			} else if imp.Native != nil {
+				return openrtb_ext.BidTypeNative, nil
+			}
+			break
+		}
+	}
+	return "", &errortypes.BadInput{
+		Message: fmt.Sprintf("Failed to find impression \"%s\"", impId),
+	}
```
Comment (onkarvhanumante) [inline adapters/gothamads/gothamads.go:174]: Warning: this is an anti-pattern. This code assumes that if there is multi-format request then mediatype is defaulting to banner. Ideally, Prebid server expects media type set in adapter response. we recommend implementing a pattern where adapter server sets the [MType](https://github.com/prebid/openrtb/blob/53ac1b9ca7ddf860988ff69bf5018b32ced1fbc5/openrtb2/bid.go#L334) field in the response to accurately determine the media type for the impression.

File: static/bidder-info/gothamads.yaml (hunk 1 above)
Comment (gotham-ads) [reply static/bidder-info/gothamads.yaml:3]: replied

File: adapters/gothamads/gothamads.go (hunk 3 above)
Comment (gotham-ads) [reply adapters/gothamads/gothamads.go:60]: ok, i've added note in https://github.com/prebid/prebid.github.io/pull/4708. Also prebid server support

File: adapters/gothamads/gothamads.go (hunk 4 above)
Comment (gotham-ads) [reply adapters/gothamads/gothamads.go:62]: it's just would be more convenient to receive request without ext

File: adapters/gothamads/gothamads.go (hunk 5 above)
Comment (gotham-ads) [reply adapters/gothamads/gothamads.go:174]: It's returning banner, video or native media else returns error. I don't use here banner as default type 
If [MType](https://github.com/prebid/openrtb/blob/53ac1b9ca7ddf860988ff69bf5018b32ced1fbc5/openrtb2/bid.go#L334) field is needed how can use it?

File: static/bidder-info/gothamads.yaml (hunk 1 above)
Comment (onkarvhanumante) [reply static/bidder-info/gothamads.yaml:3]: received response

File: adapters/gothamads/gothamads.go (hunk 5 above)
Comment (gotham-ads) [reply adapters/gothamads/gothamads.go:174]: @onkarvhanumante it's easy to change to MType but we're working with [openrtb 2.5 specification](https://www.iab.com/wp-content/uploads/2016/03/OpenRTB-API-Specification-Version-2-5-FINAL.pdf) which doesn't support mtype field. But if needed we can add a mtype from bidder side

Comment (onkarvhanumante) [reply adapters/gothamads/gothamads.go:174]: latest release of PBS has support for ortb 2.6 specs. so it is advisable to use it

Comment (gotham-ads) [reply adapters/gothamads/gothamads.go:169]: fixed

File: adapters/gothamads/gothamads_test.go [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+	"github.com/prebid/prebid-server/adapters/adapterstest"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestJsonSamples(t *testing.T) {
+	bidder, buildErr := Builder(openrtb_ext.BidderGothamads, config.Adapter{
+		Endpoint: "http://us.example.com/?pass={{.AccountID}}"}, config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "2"})
```
Comment (onkarvhanumante) [inline adapters/gothamads/gothamads_test.go:14]: just as a practise. lets match endpoint with `http://us-e-node1.gothamads.com/?pass={{.AccountID}}` mentioned in   bidder-info yaml

**Review by Sonali-More-Xandr — DISMISSED** [review]

File: adapters/gothamads/gothamads_test.go (hunk 6 above)
Comment (gotham-ads) [reply adapters/gothamads/gothamads_test.go]: fixed

**Review by onkarvhanumante — APPROVED** [review]

**Review by Sonali-More-Xandr — APPROVED** [review]

**Pull request merged**